### ISSUE TITLE
plexus-compiler-eclipse: some warnings from APT processors are lost

### ISSUE THREAD

**Issue by tomaswolf** [codehaus-plexus/plexus-compiler#68]
See [Eclipse bug 559645](https://bugs.eclipse.org/bugs/show_bug.cgi?id=559645) for context and a sample Eclipse maven project showing the issue.

Basically, ECJ reports these diagnostics from APT processors as "extra_problem" elements in the log XML. plexus-compiler-eclipse completely ignores these when parsing the XML to get the diagnostics. Even if it _did_ parse them, ECJ at least up to 3.20.0 has a bug and doesn't report the source file for these extra_problems in the XML, so the output would not be really helpful.

**Comment by tomaswolf** [comment]
See PR #69.

**Comment by tomaswolf** [comment]
Thanks for merging. What's going on with the Travis builds? jdk8/9 fails consistently, jdk11 succeeds.

I suppose this issue could be closed since the PR is merged. When could we expect a release with this?

**Comment by olamy** [comment]
switching to openjdk fix the problem

**Comment by olamy** [comment]
release done :)

**Comment by tomaswolf** [comment]
Thanks. Just built a fairly large Spring Boot project with this; works well.

In fact, it works too well: it emits the missing warnings, but also shows a few other warnings that previously were hidden, in particular
```
[WARNING] Unknown source: Field value processing of @ConfigurationProperty meta-data is not supported
[WARNING] Unknown source: Hibernate JPA 2 Static-Metamodel Generator 5.3.7.Final
```
Both are also visible with plexus-compiler-eclipse 2.8.5 if maven is run with -X.

Don't know where exactly the first one comes from...

The second one is actually another ECJ bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=560204 . Hibernate [emits this as an INFO](https://github.com/hibernate/hibernate-metamodelgen/blob/f06a5e1/src/main/java/org/hibernate/jpamodelgen/JPAMetaModelEntityProcessor.java#L87), but ECJ reports any non-error message as a warning.

**Comment by tomaswolf** [comment]
While https://bugs.eclipse.org/bugs/show_bug.cgi?id=560204 is fixed in the upcoming ECJ 3.22.0, the switch to JSR-199 also exposes another ECJ bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=563654 : sometimes ECJ reports suppressed warnings. A fix is pending in ECJ at https://git.eclipse.org/r/#/c/163738/ .

An there's follow-up issue #73 (configured encoding not respected), for which there is PR #74 pending here.
